terraform/dns/main.tf

```diff
@@ -0,0 +1,229 @@
+terraform {
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = "5.17.0"
+    }
+
+    dnsimple = {
+      source  = "dnsimple/dnsimple"
+      version = "1.2.1"
+    }
+  }
+}
+
+provider "aws" {
+  region = "us-east-1"
+}
+
+provider "aws" {
+  alias  = "aws-sydney"
+  region = "ap-southeast-2"
+}
+
+provider "dnsimple" {
+}
+
+variable "domain" {
+  type = string
+}
+
+// To demonstrate MX and TXT records and how you may use these with mail services, we'll set up AWS SES for our domain. First, we'll verify and set up our domain entirely within Terraform — no manual copy-and-pasting necessary. Then, we'll do some basic SES configuration to store incoming emails to our domain into an S3 bucket, so we can actually do something useful with this mail service setup.
+
+// Create our domain with SES. It won't be verified yet — we'll do that next.
+resource "aws_ses_domain_identity" "domain" {
+  domain = var.domain
+}
+
+// We need to verify our domain, so SES knows we are the true owners of the domain. This involves a simple TXT record, which we can set using the DNSimple zone record resource.
+resource "dnsimple_zone_record" "verification_record" {
+  zone_name = var.domain
+  name      = "_amazonses"
+  type      = "TXT"
+  ttl       = 600
+  value     = aws_ses_domain_identity.domain.verification_token
+}
+
+// Now that the record has been set, we'll ask SES to verify and domain and wait for its confirmation. This may take a few minutes...
+resource "aws_ses_domain_identity_verification" "domain" {
+  domain     = aws_ses_domain_identity.domain.id
+  depends_on = [dnsimple_zone_record.verification_record]
+}
+
+// For this demo, we'll simply store all incoming emails in our S3 bucket, so lets create one now. SES can do more with incoming emails; see https://docs.aws.amazon.com/ses/latest/dg/receiving-email-concepts.html for more details.
+resource "aws_s3_bucket" "emails" {
+  bucket = "${var.domain}-emails"
+}
+
+// This sets the policy for the S3 bucket we've just created to allow SES to write to it. See more details at https://docs.aws.amazon.com/ses/latest/dg/receiving-email-permissions.html#receiving-email-permissions-s3.
+resource "aws_s3_bucket_policy" "emails" {
+  bucket = aws_s3_bucket.emails.id
+  policy = jsonencode(templatefile("iam/emails-bucket-policy.json", {
```

Comment:
`jsonencode` should be removed for all `templatefile` and `file` instances, as it produces invalid encoding of JSON string instead of JSON object.